server/src/main/java/org/elasticsearch/cluster/coordination/Join.java

```diff
@@ -16,59 +16,65 @@
 
 /**
  * Triggered by a {@link StartJoinRequest}, instances of this class represent join votes,
- * and have a source and target node. The source node is the node that provides the vote,
- * and the target node is the node for which this vote is cast. A node will only cast
- * a single vote per term, and this for a unique target node. The vote also carries
- * information about the current state of the node that provided the vote, so that
- * the receiver of the vote can determine if it has a more up-to-date state than the
- * source node.
+ * and have a voting and candidate-master node. The voting node is the node that provides
+ * the vote, and the candidate-master node is the node for which this vote is cast. A join
+ * vote is cast to reform the cluster around a particular master-eligible node, to elect
+ * that node as the new master in a new term.
+ *
+ * A voting node will only cast a single vote per term. The vote includes information about
+ * the current state of the node casting the vote, so that the candidate for the vote can
+ * determine whether it has a more up-to-date state than the voting node.
  */
 public class Join implements Writeable {
```

Comment:
NIT: I think this could become a record

Reply:
Sounds good 👍  I went ahead and did it, but there's a lot of code churn due to renaming all the get*() methods to the record default method names. I'll put up a separate PR once this lands, if that's alright.